### PR TITLE
provider/azurerm: Add example of a Linux VM w/ Managed Disks

### PR DESCRIPTION
**Pull request by scottzilla**
**Manual merge required; we have modified the .travis.yml in our fork to do CI of examples.**

This example provisions and deploys the following resources into an Azure subscription:

Resource Group
VNet
Subnet
NIC
PIP
Storage Account (for bootDiagnostics)
OS Disk (Managed)
1TB Data Disk (Managed)

Passing CI process: https://travis-ci.org/harijayms/terraform/builds/226973470

```
terraform get, validate, plan to output, apply output, show
validate with azure CLI
terraform destroy
```

### CODE REVIEW

File: examples/azure-vm-simple-linux-managed-disk/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Very simple deployment of a Linux VM
+
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vm-simple-linux%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/README.md]: Is it possible to link to a Terraform version of this, rather than an ARM template (which coming from the `master` branch could get out of sync)?

File: examples/azure-vm-simple-linux-managed-disk/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+output "hostname" {
+  value = "${var.hostname}"
+}
+
+output "vm_fqdn" {
+  value = "${azurerm_public_ip.pip.fqdn}"
+}
+
+output "sshCommand" {
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/outputs.tf]: to match the terraform naming conventions, it might be best to name this `ssh_command`?

File: .travis.yml [hunk 3]
```diff
@@ -1,42 +1,33 @@
-dist: trusty
-sudo: false
-language: go
-go:
-- 1.8.x
+####################################################################################
+## NOT FOR UPSTREAM PROPOSAL; INTENDED FOR CI OF AZURE EXAMPLES IN THIS REPO ONLY ##
+####################################################################################
```
Comment (tombuildsstuff) [inline .travis.yml]: once these issues have been addressed - is it possible to revert the changes to this file?

File: examples/azure-vm-simple-linux-managed-disk/main.tf [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = "${azurerm_public_ip.pip.id}"
+  }
+}
+
+resource "azurerm_public_ip" "pip" {
+  name                         = "${var.rg_prefix}-ip"
+  location                     = "${var.location}"
+  resource_group_name          = "${azurerm_resource_group.rg.name}"
+  public_ip_address_allocation = "dynamic"
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/main.tf]: is it worth making this `Dynamic` to match the field above?

File: examples/azure-vm-simple-linux-managed-disk/terraform.tfvars [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Replace with relevant values
+
+# resource_group = "myresourcegroup"
+# rg_prefix      = "rg"
+# hostname       = "myvm"
+# dns_name       = "mydnsname"
+# location       = "southcentralus"
+# admin_password = "T3rr@f0rmP@ssword"
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/terraform.tfvars]: Would it be possible to set these as defaults in the `variables.tf` file and then remove this file?

It's also worth mentioning if there's no default in Code, the user will be prompted at runtime - which might be a better fit for the sensitive variables e.g. `admin_password`?

File: examples/azure-vm-simple-linux-managed-disk/provider.tf [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# provider "azurerm" {
+#   subscription_id = "REPLACE-WITH-YOUR-SUBSCRIPTION-ID"
+#   client_id       = "REPLACE-WITH-YOUR-CLIENT-ID"
+#   client_secret   = "REPLACE-WITH-YOUR-CLIENT-SECRET"
+#   tenant_id       = "REPLACE-WITH-YOUR-TENANT-ID"
+# }
+
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/provider.tf]: is it possible to move this into the top of the `main.tf` file?

File: examples/azure-vm-simple-linux-managed-disk/main.tf [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    admin_username = "${var.admin_username}"
+    admin_password = "${var.admin_password}"
+  }
+
+  os_profile_linux_config {
+    disable_password_authentication = false
+  }
+
+  boot_diagnostics {
+    enabled     = "true"
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/main.tf]: I think we can remove the quotes here?

File: examples/azure-vm-simple-linux-managed-disk/README.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+This template allows you to deploy a simple Linux VM using a few different options for the Ubuntu version, using the latest patched version. This will deploy a A1 size VM in the resource group location and return the FQDN of the VM.
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/README.md]: This deploys an `A0` size VM?

File: examples/azure-vm-simple-linux-managed-disk/README.md (hunk 1 above)
Comment (scottzilla) [inline examples/azure-vm-simple-linux-managed-disk/README.md]: Sure, we can link to our fork's `master`.

File: examples/azure-vm-simple-linux-managed-disk/main.tf (hunk 4 above)
Comment (scottzilla) [inline examples/azure-vm-simple-linux-managed-disk/main.tf]: Sure, we can update it since we're making other changes.

File: examples/azure-vm-simple-linux-managed-disk/terraform.tfvars (hunk 5 above)
Comment (scottzilla) [inline examples/azure-vm-simple-linux-managed-disk/terraform.tfvars]: Sure @tombuildsstuff, we'll remove this file altogether, set defaults in `variables.tf`, and leave sensitive variables (`admin_password`) without a default. 👍

File: examples/azure-vm-simple-linux-managed-disk/main.tf (hunk 7 above)
Comment (scottzilla) [inline examples/azure-vm-simple-linux-managed-disk/main.tf]: Yea, I think so too.

File: examples/azure-vm-simple-linux-managed-disk/README.md (hunk 1 above)
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/README.md]: @scottzilla sorry, to clarify that: is there a similar tool for displaying Terraform configuration files, rather than an ARM Template? If not, I think it'd make sense to remove this, rather than confuse users?
